targets/TARGET_Cypress/TARGET_PSOC6/CMakeLists.txt

```diff
@@ -46,14 +46,16 @@ target_sources(mbed-cm0p-secure
         psoc6cm0p/COMPONENT_CM0P_SECURE/psoc6_03_cm0p_secure.c
 )
 
-add_library(mbed-cm0p-sleep INTERFACE)
-target_sources(mbed-cm0p-sleep
-    INTERFACE
-        psoc6cm0p/COMPONENT_CM0P_SLEEP/psoc6_01_cm0p_sleep.c
-        psoc6cm0p/COMPONENT_CM0P_SLEEP/psoc6_02_cm0p_sleep.c
-        psoc6cm0p/COMPONENT_CM0P_SLEEP/psoc6_03_cm0p_sleep.c
-        psoc6cm0p/COMPONENT_CM0P_SLEEP/psoc6_04_cm0p_sleep.c
-)
+if("CM0P_SLEEP" IN_LIST MBED_TARGET_LABELS)
+    add_library(mbed-cm0p-sleep INTERFACE)
+    target_sources(mbed-cm0p-sleep
+        INTERFACE
+            psoc6cm0p/COMPONENT_CM0P_SLEEP/psoc6_01_cm0p_sleep.c
+            psoc6cm0p/COMPONENT_CM0P_SLEEP/psoc6_02_cm0p_sleep.c
+            psoc6cm0p/COMPONENT_CM0P_SLEEP/psoc6_03_cm0p_sleep.c
+            psoc6cm0p/COMPONENT_CM0P_SLEEP/psoc6_04_cm0p_sleep.c
+    )
```

Comment:
My CMake knowledge is underdeveloped. Should I read this to mean that the mbed-cm0p-sleep will include all four of these C files? That shouldn't be the case, as each of them is a unique main() (pre-built and loaded into a C array). If that is correct, then some boards are not going to get the correct version of the main() and may not behave correctly.

Reply:
It copies the same structure as with mbed-cli 1. There are ifdef in place in each file `ifdef CY_DEVICE_PSOC6A256K` and similar.

All files are included in the built with cli 1 or 2, but only one gets compiled based on target selection.

Reply:
Excellent, forgot that point. Thank you. Cypress/Infineon approval was not asked for, but I approve nonetheless.